hpedockerplugin/volume_manager.py

```diff
@@ -1837,7 +1867,13 @@ def _rollback(rollback_list):
         for undo_action in reversed(rollback_list):
             LOG.info(undo_action['msg'])
             try:
-                undo_action['undo_func'](**undo_action['params'])
+                params = undo_action['params']
+                if type(params) is dict:
+                    undo_action['undo_func'](**undo_action['params'])
```

Comment:
Any specific reason to identify the data type of the `params` and call the undo functions differently.. I'm asking due to the previous code does'nt have this kind of check ?

Reply:
i got the answer. Tuple is passed explicitly in `undo_steps.append(..'params')`
```
undo_steps.append(
                    {'undo_func': driver.terminate_connection,
                     'params': (vol, connector_info, is_snap),
```